frontend/packages/dev-console/src/components/topology/shapes/PodSet.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import { PodStatus, calculateRadius, getPodData } from '@console/shared';
+import { DonutStatusData } from '../topology-types';
+
+interface PodSetProps {
+  size: number;
+  data: DonutStatusData;
+}
+
+interface InnerPodStatusRadius {
+  innerPodStatusOuterRadius: number;
+  innerPodStatusInnerRadius: number;
+}
+
+const calculateInnerPodStatusRadius = (
+  outerPodStatusInnerRadius: number,
+  outerPodStatusWidth: number,
+): InnerPodStatusRadius => {
+  const innerPodStatusWidth = outerPodStatusWidth * 0.6;
+  const spaceBwOuterAndInnerPodStatus = 3;
+  const innerPodStatusOuterRadius = outerPodStatusInnerRadius - spaceBwOuterAndInnerPodStatus;
+  const innerPodStatusInnerRadius = innerPodStatusOuterRadius - innerPodStatusWidth;
+
+  return { innerPodStatusOuterRadius, innerPodStatusInnerRadius };
+};
+
+const PodSet: React.FC<PodSetProps> = ({ size, data }) => {
+  const { podStatusOuterRadius, podStatusInnerRadius, podStatusStrokeWidth } = calculateRadius(
+    size,
+  );
+  const { innerPodStatusOuterRadius, innerPodStatusInnerRadius } = calculateInnerPodStatusRadius(
```

Comment:
Was going to suggest we reuse `calculateRadius` (instead of creating `calculateInnerPodStatusRadius`) and make it accept an optional `strokeWidth` - but that code makes me uncomfortable lol. I think we are best writing tests around it and then worrying about consolidating that logic.

Reply:
I created https://jira.coreos.com/browse/ODC-2012 to track the above change.